### DISCUSSION TITLE
Koneksi Jupyter Notebook Ke ODBC

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#138] [Q&A] [answered]
Bagaimana cara membuat koneksi ODBC di jupyter notebook. berikut ada beberapa data ODBC yang terkoneksi.

<img width="442" alt="image" src="https://user-images.githubusercontent.com/103427989/174291256-f631dc6a-7e08-47d4-b583-0e623e781639.png">

**Comment by fafilia** [comment, marked as answer]
Untuk bisa connect dengan ODBC, kita bisa menggunakan library pyodbc sebagai dbapi penghubung dengan python. Ada 2 langkah yang umum yang perlu dipersiapkan, yaitu:
1. Membangun koneksi agar python terhubung dengan ODBC
2. Fetching data yang akan diread menggunakan pandas

Berikut adalah contoh dari proses fetching data dari ODBC

```
import pyodbc
import pandas as pd

#code di bawah ini untuk membangun koneksi ke ODBC
cnxn = pyodbc.connect(r'DRIVER={Microsoft Access Driver (*.mdb, *.accdb)};'
                      r'DBQ=C:\users\bartogre\desktop\data.mdb;')

# code dibawah ini untuk fetching data
sql = "Select sum(CYTM), sum(PYTM), BRAND From data Group By BRAND"
data = pd.read_sql(sql,cnxn)  # without parameters [non-prepared statement]
```

Referensi terkait [ODBC](https://stackoverflow.com/questions/39835770/read-data-from-pyodbc-to-pandas)

**Reply by joejnt** [reply]
Dear @fafilia 

Terima kasih untuk insightnya.

Best regards,
JT
